### ISSUE TITLE
Document units for polyfit when dimension is time

### ISSUE THREAD

**Issue by andrewpauling** [pydata/xarray#4455]
**Is your feature request related to a problem? Please describe.**
I think (please correct me if I'm wrong) when using polyfit with dim='time' the units of the output slope are [data_units]/ns, but this not explained in the docstring or in the documentation on the webpage. I figured this out eventually but it could be confusing for new users.

**Describe the solution you'd like**
Mention in the documentation and/or docstring that the units for time will be ns

**Comment by navidcy** [comment]
Yes please! This is important to note. We were struggling a bit to figure it out :) 

cc: @lidefi87, @paigem

**Comment by navidcy** [comment]
Well on second thought the units were there written in `datetim64[ns]`. I became skeptical regarding whether a note was needed so I closed the PR. Happy to hear thoughts.
